pkg/compact/compact.go

```diff
@@ -194,6 +197,9 @@ func (c *Syncer) syncMetas(ctx context.Context) error {
 					continue
 				}
 				if err != nil {
+					if removed := c.removeIfMalformed(workCtx, id); removed {
```

Comment:
all of this is great but look 2 lines above. We already have `blockTooFreshSentinelError` and stuff that checks if is fresh enough! Let's reuse exactly the same thing if we can!

Reply:
I've changed it up so that in either case where the block is too fresh this is reported by `DownloadMeta`.  The logic is slightly different based on whether or not we were able to get a meta for that block, but it's nicer to have only one place doing the freshness check.